include/openenclave/internal/sgxsign.h

```diff
@@ -35,6 +35,31 @@ oe_result_t oe_sgx_sign_enclave(
     size_t pem_size,
     sgx_sigstruct_t* sigstruct);
 
+/**
+ * Digitally sign the enclave with the given hash using an openssl engine
+ *
+ * This function digitally signs the enclave whose hash is given by the
+ * **mrenclave** parameter. The signing key is given by the **engine_id** and related **key_id**
+ * parameter. If successful, the function writes the signature into the
+ * **sigstruct** parameter (an SGX signature structure).
+ *
+ * @param mrenclave[in] hash of the enclave being signed
+ * @param engine_id[in] Text name of the engine to use
+ * @param key_id[in]    Integer handle for the key to use.
+ * @param sigstruct[out] the SGX signature
+ *
+ * @return OE_OK success
+ */
+oe_result_t oe_sgx_sign_enclave_from_engine(
+    const OE_SHA256* mrenclave,
+    uint64_t attributes,
+    uint16_t product_id,
+    uint16_t security_version,
+    const char *engine_id,
+    const char *engine_load_path,
+    const char * key_id,
```

Comment:
nit: inconsistant spacing around *.  Compare line 54, line 59, and 60, which all use different styles.  Pick one.   (Personally I would be in favor of using something like uncrustify to do such things automatically)

Reply:
done